logins-sql/src/schema.rs

```diff
@@ -0,0 +1,202 @@
+    -- Milliseconds
+    timeLastUsed        INTEGER,
+    -- Milliseconds
+    timePasswordChanged INTEGER NOT NULL,
+    username            TEXT,
+    password            TEXT NOT NULL,
+    guid                TEXT NOT NULL UNIQUE
+";
+
+
+
+lazy_static! {
+    static ref CREATE_LOCAL_TABLE_SQL: String = format!(
+        "CREATE TABLE IF NOT EXISTS {local} (
+            {common_sql},
+            -- Milliseconds
+            local_modified INTEGER,
+
+            is_deleted     TINYINT NOT NULL DEFAULT 0,
+            sync_status    TINYINT NOT NULL DEFAULT 0
+        )",
+        local      = LOCAL_TABLE_NAME,
+        common_sql = COMMON_SQL
+    );
+
+    static ref CREATE_MIRROR_TABLE_SQL: String = format!(
+        "CREATE TABLE IF NOT EXISTS {mirror} (
+            {common_sql},
+            server_modified INTEGER NOT NULL,
+            is_overridden   TINYINT NOT NULL DEFAULT 0
+        )",
+        mirror     = MIRROR_TABLE_NAME,
+        common_sql = COMMON_SQL
+    );
+
+    static ref CREATE_META_TABLE_SQL: String = format!(
+        "CREATE TABLE IF NOT EXISTS {meta} (
+            key TEXT PRIMARY KEY,
+            value NOT NULL
+        )",
+        meta = META_TABLE_NAME,
+    );
+
+    static ref CREATE_OVERRIDE_HOSTNAME_INDEX_SQL: String = format!(
+        "CREATE INDEX IF NOT EXISTS {idx_override_hostname} ON {mirror} (is_overridden, hostname)",
+        idx_override_hostname = IDX_MIRROR_OVERRIDEN_HOSTNAME,
+        mirror = MIRROR_TABLE_NAME
+    );
+
+    static ref CREATE_DELETED_HOSTNAME_INDEX_SQL: String = format!(
+        "CREATE INDEX IF NOT EXISTS {idx_local_deleted_hostname} ON {local} (is_deleted, hostname)",
+        idx_local_deleted_hostname = IDX_LOCAL_DELETED_HOSTNAME,
+        local = LOCAL_TABLE_NAME
+    );
+
+    static ref UPDATE_LOCAL_TIMESTAMPS_TO_MILLIS_SQL: String = format!(
+        "UPDATE {local}
+         SET timeCreated = timeCreated / 1000,
+             timeLastUsed = timeLastUsed / 1000,
+             timePasswordChanged = timePasswordChanged / 1000",
+        local = LOCAL_TABLE_NAME
+    );
+
+    static ref UPDATE_MIRROR_TIMESTAMPS_TO_MILLIS_SQL: String = format!(
+        "UPDATE {mirror}
+         SET timeCreated = timeCreated / 1000,
+             timeLastUsed = timeLastUsed / 1000,
+             timePasswordChanged = timePasswordChanged / 1000",
+        mirror = MIRROR_TABLE_NAME
+    );
+
+    static ref SET_VERSION_SQL: String = format!(
+        "PRAGMA user_version = {version}",
+        version = VERSION
+    );
+}
+
+pub(crate) static LAST_SYNC_META_KEY:    &'static str = "last_sync_time";
+pub(crate) static GLOBAL_STATE_META_KEY: &'static str = "global_state";
+
+pub fn init(db: &db::LoginDb) -> Result<()> {
+    let user_version = db.query_one::<i64>("PRAGMA user_version")?;
+    if user_version == 0 {
+        let table_list_exists = db.query_one::<i64>(
+            "SELECT count(*) FROM sqlite_master WHERE type = 'table' AND name = 'tableList'"
```

Comment:
I think I understand why you're doing this -- maybe? -- but it feels like you're only going half way here.  That is you're using user version, but not actually checking for exactly the table name(s) that are relevant.  Explain what I'm missing?

Reply:
To be honest, this is mostly just mimicing the logic from https://github.com/mozilla-lockbox/lockbox-ios-fxa-sync/blob/master/Storage/ThirdParty/SwiftData.swift#L775